packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -111,16 +111,28 @@ abstract class AotAssemblyBase extends Target {
     if (results.any((int result) => result != 0)) {
       throw Exception('AOT snapshotter exited with code ${results.join()}');
     }
-    final String resultPath = environment.fileSystem.path.join(environment.buildDir.path, 'App.framework', 'App');
+
+    await _lipo(environment, darwinArchs, 'App.Framework/App', buildOutputPath);
+
+    if (buildMode == BuildMode.profile || buildMode == BuildMode.release) {
```

Comment:
The logic of when the dSYM is present is now sprinkled around in a few places.  Would be better for `_lipo` to return if the path doesn't exist.  It's `lipo`ing mostly for x64 and arm64 simulators now, but not that long ago we supported armv7 and arm64 physical devices as well.

Reply:
changed to check for file presence